### PR TITLE
Fix/meta 3919 - Check for status of task from Graph store also

### PR DESCRIPTION
**Pull request by sumandas0**
## Add check for status of task from Graph store also

> Recently, we released new code into production. However, we identified an issue where a certain part of the code needed improvement. Specifically, we were checking for the existence of pending tasks of a certain classification id and, if found, not scheduling the task. However, this was being done by fetching information from Elasticsearch (ES), which can lead to inconsistencies.o address this, we added a fix to also check the status of the task in Graph store, ensuring accuracy and consistency.

### CODE REVIEW

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java [hunk 1]
```diff
@@ -1376,35 +1377,38 @@ public void createAndQueueClassificationRefreshPropagationTask(AtlasEdge edge) t
...
+                return true;
             }
-        }
 
-        //Search in ES to get the tasks
-        TaskSearchResult taskSearchResult = taskUtil.findPendingTasksByClassificationId(0, 1, classificationId, new ArrayList<String>(){{
-            add(CLASSIFICATION_PROPAGATION_DELETE);
-            add(CLASSIFICATION_REFRESH_PROPAGATION);
-        }}, new ArrayList<>());
+            TaskSearchResult taskSearchResult = taskUtil.findPendingTasksByClassificationId(0, PENDING_TASK_QUERY_SIZE_LIMIT,
```
Comment (mehtaanshul) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java:1395]: Add an info log, if we get more than 1 task

File: repository/src/main/java/org/apache/atlas/repository/graph/GraphHelper.java [hunk 2]
```diff
@@ -393,25 +393,28 @@ public static AtlasEdge getClassificationEdge(AtlasVertex entityVertex, AtlasVer
...
+                while (vertices.hasNext()) {
+                    AtlasVertex vertex = vertices.next();
+                    if (vertex != null) {
+                        if (vertex.getIdForDisplay().equals(classificationId)) {
+                            return true;
+                        }
                     }
                 }
             }
+        } catch (Exception e) {
```
Comment (mehtaanshul) [inline repository/src/main/java/org/apache/atlas/repository/graph/GraphHelper.java:413]: Why not throw an error?

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java [hunk 3]
```diff
@@ -1376,35 +1377,38 @@ public void createAndQueueClassificationRefreshPropagationTask(AtlasEdge edge) t
...
-        List<AtlasTask> queuedTasks  = RequestContext.get().getQueuedTasks();
-        for(AtlasTask queuedTask : queuedTasks) {
-            if(queuedTask.getClassificationId().equals(classificationId)) {
-                String queuedTaskType = queuedTask.getType();
-                if(queuedTaskType.equals(CLASSIFICATION_REFRESH_PROPAGATION)
-                        || queuedTaskType.equals(CLASSIFICATION_PROPAGATION_DELETE)) {
-                    return true;
-                }
+            List<String> taskTypes = Arrays.asList(CLASSIFICATION_REFRESH_PROPAGATION, CLASSIFICATION_PROPAGATION_DELETE);
+            if (RequestContext.get().getQueuedTasks().stream().anyMatch(task -> task.getClassificationId().equals(classificationId)
```
Comment (mehtaanshul) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java:1390]: null check

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java [hunk 4]
```diff
@@ -1376,35 +1377,38 @@ public void createAndQueueClassificationRefreshPropagationTask(AtlasEdge edge) t
...
-            add(CLASSIFICATION_PROPAGATION_DELETE);
-            add(CLASSIFICATION_REFRESH_PROPAGATION);
-        }}, new ArrayList<>());
+            TaskSearchResult taskSearchResult = taskUtil.findPendingTasksByClassificationId(0, PENDING_TASK_QUERY_SIZE_LIMIT,
+                    classificationId, taskTypes , new ArrayList<>());
 
-        if (!taskSearchResult.getTasks().isEmpty()) {
-            return true;
+            // if any task have status as PENDING, then skip task creation
+            if (taskSearchResult.getTasks().stream().anyMatch(task -> task.getClassificationId().equals(classificationId)
```
Comment (mehtaanshul) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java:1399]: null check

File: repository/src/main/java/org/apache/atlas/repository/graph/GraphHelper.java (hunk 2 above)
Comment (sumandas0) [reply repository/src/main/java/org/apache/atlas/repository/graph/GraphHelper.java]: Yeah, have to throw error

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java (hunk 4 above)
Comment (sumandas0) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java]: Added

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java (hunk 3 above)
Comment (sumandas0) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java]: Added

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java (hunk 1 above)
Comment (sumandas0) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java:1399]: There could be 2 task sometime if delete task is there, so added a check like if more than one refresh classification task is there adding a warn log

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java [hunk 5]
```diff
@@ -1387,20 +1387,43 @@ private boolean skipClassificationTaskCreation(String classificationId) throws A
         try {
 
             List<String> taskTypes = Arrays.asList(CLASSIFICATION_REFRESH_PROPAGATION, CLASSIFICATION_PROPAGATION_DELETE);
-            if (RequestContext.get().getQueuedTasks().stream().anyMatch(task -> task.getClassificationId().equals(classificationId)
-                    && taskTypes.contains(task.getType()))) {
+            if (
+                    RequestContext.get().getQueuedTasks().stream()
```
Comment (mehtaanshul) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java:1391]: Add null check for "RequestContext.get().getQueuedTasks()"

Comment (sumandas0) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java:1391]: getQueuedTasks will always return empty array not null

Comment (sumandas0) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java]: Added the check

**Pull request merged**